### PR TITLE
[SPARK-10458] [Spark Core] Added isStopped() method in SparkContext

### PR DESCRIPTION
**Pull request by kmadhugit**
Added isStopped() method in SparkContext

### CODE REVIEW

**Comment by srowen** [comment]
That seems reasonable, as long as its semantics are understood: stopping has begun, not necessarily completed. I'd like at least one other +1 before proceeding just because it's an API addition in the core.

**Comment by holdenk** [comment]
Maybe since the semantics are a bit difficult to tell from the name (e.g. stopped includes stopping which wouldn't be my default assumption) adding scaladoc for the semantics might be a good idea?

**Comment by andrewor14** [comment]
ok to test

**Comment by andrewor14** [comment]
Yes I agree @kmadhugit can you add a java doc to explain the semantics?

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 1]
```diff
@@ -264,6 +264,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   val tachyonFolderName = externalBlockStoreFolderName
 
   def isLocal: Boolean = (master == "local" || master.startsWith("local["))
+  def isStopped: Boolean = stopped.get()
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: can we add some javadoc for this

Comment (tedyu) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: Should this be annotated @DeveloperApi ?

Comment (rxin) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: It seems to me it should be fine to be public, since we can always have isStopped

Comment (tedyu) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: HiveThriftServer2.scala can reference isStopped now, right ?

**Comment by kmadhugit** [comment]
This failure has nothing to do with my patch, may be a retest would help?

**Comment by srowen** [comment]
Yeah, the failures are spurious. If you're bored waiting, any trivial test for this (maybe piggy-backing on an existing test logic) would be good, just to make sure it does what's expected. But yes LGTM and the test failures are unrelated.

**Comment by andrewor14** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 2]
```diff
@@ -264,6 +264,10 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   val tachyonFolderName = externalBlockStoreFolderName
 
   def isLocal: Boolean = (master == "local" || master.startsWith("local["))
+  /**
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: need space above this. I've added it myself

**Comment by andrewor14** [comment]
Merging into master.

**Pull request closed without merging**